bundles/org.eclipse.sirius.emfjson/src/main/java/org/eclipse/sirius/emfjson/utils/GsonEObjectSerializer.java

```diff
@@ -1040,22 +1043,48 @@ private JsonElement serializeEDataType(EObject eObject, EAttribute eAttribute) {
             if (value instanceof Collection) {
                 Collection<?> collection = (Collection<?>) value;
                 for (Object object : collection) {
-                    jsonArray.add(new JsonPrimitive(eFactoryInstance.convertToString(eAttribute.getEAttributeType(), object)));
+                    if (object == null) {
+                        jsonArray.add(JsonNull.INSTANCE);
+                    } else if (this.isEDataTypeSerializableInJson(eAttribute.getEAttributeType())) {
+                        jsonArray.add(new Gson().toJsonTree(object));
+                    } else {
+                        jsonArray.add(new JsonPrimitive(eFactoryInstance.convertToString(eAttribute.getEAttributeType(), object)));
+                    }
                 }
             }
             jsonElement = jsonArray;
         } else {
             String stringValue = eFactoryInstance.convertToString(eAttribute.getEAttributeType(), value);
-
-            if (stringValue == null) {
-                stringValue = ""; //$NON-NLS-1$
+            if (this.isEDataTypeSerializableInJson(eAttribute.getEAttributeType())) {
+                jsonElement = new Gson().toJsonTree(value);
+            } else {
+                if (stringValue == null) {
```

Comment:
prefer checking null first like line 1046. It avoid letting the given Predicate manage this case (if any)

Reply:
Yes you're right, the code will be cleaner like that. I fix that right now.

Reply:
Done.